src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,33 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim.hidden === false) {
+      victim.health -= 50;
+
+      if (victim.health <= 0) {
+        Animal.alive = Animal.alive.filter((creature) => creature.health > 0);
```

Comment:
[CODE PERFORMANCE] - Instead of checking the health of the victim after each bite, consider removing dead animals from the `Animal.alive` array in a separate method. This would avoid the need to filter the array every time the `bite` method is called, which could be more efficient if the `bite` method is called frequently.

Reply:
> [CODE PERFORMANCE] - Instead of checking the health of the victim after each bite, consider removing dead animals from the `Animal.alive` array in a separate method. This would avoid the need to filter the array every time the `bite` method is called, which could be more efficient if the `bite` method is called frequently.

Good point, you can create a separate method for the animal class, so you can reuse it in other cases